### PR TITLE
fix: Panic on CJK character truncation in MCP prompt descriptions

### LINKED ISSUE #3117: Amazon Q CLI crashes when there is CJK MCP Prompts description
### Checks

- [x] I have searched [github.com/aws/amazon-q-developer-cli/issues](https://github.com/aws/amazon-q-developer-cli/issues?q=) and there are no duplicates of my issue
- [x] I have run `q doctor` in the affected terminal session
- [x] I have run `q restart` and replicated the issue again

### Operating system

macOS 26.0.0 (25A354)

### Expected behaviour

Amazon Q Developer CLI should not crash when there is Korean MCP Prompt description.

### Actual behaviour

Amazon Q Developer CLI crashes when there is long MCP Prompt description written in CJK languages.

### Steps to reproduce

* install mcp prompt server that serves MCP Prompt written in CJK languages
* execute `/prompts` or `/prompts  list`
* Amazon Q Developer CLI crashes with following error
```
The application panicked (crashed).
Message:  byte index 37 is not a char boundary; it is inside '한' (bytes 36..39) of `사용자가 작성한 글의 어색한 표현이나 오타를 수정하고 싶을 때 사용할 수 있는 프롬프트로, 원본의 어투와 어조를 유지하면서 자연스러운 표현으로
 개선해줍니다`
Location: crates/chat-cli/src/cli/chat/cli/prompts.rs:248
```

### Environment

```yaml
<This will be visible to anyone. Do not include personal or sensitive information>

[q-details]
version = "1.17.1"
hash = "b0ef39f91625c36c2ed6f065628e7165fae9012c"
date = "2025-09-30T22:01:25.726592Z (6d ago)"

[system-info]
os = "macOS 26.0.0 (25A354)"
chip = "Apple M1 Pro"
total-cores = 8
memory = "16.00 GB"
```

### LINKED ISSUE #3170: Panic when byte index is not a char boundary on non-ASCII strings
### Checks

- [x] I have searched [github.com/aws/amazon-q-developer-cli/issues](https://github.com/aws/amazon-q-developer-cli/issues?q=) and there are no duplicates of my issue
- [x] I have run `q doctor` in the affected terminal session
- [x] I have run `q restart` and replicated the issue again

### Operating system

macOS 26.0.1 (25A362)

### Expected behaviour

Q CLI 能夠正確處理包含中文或其他多位元 UTF-8 字元的輸入文字，而不會發生 panic 或 crash。訊息應該成功送出並回傳正常回應。

### Actual behaviour

當輸入包含中文時，Q CLI 發生 panic 並 crash，錯誤訊息如下：
The application panicked (crashed).
Message: byte index 60 is not a char boundary; it is inside '成' (bytes 59..62) of `移除 eagleeye-ec-databases 任務狀況確認…`
Location: crates/chat-cli/src/cli/chat/checkpoint.rs:373
Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

### Steps to reproduce

1. 安裝並開啟 Amazon Q CLI（版本 x.y.z）。  
2. 在終端機執行以下命令，將引號內的文字替換為包含中文的訊息：  
   ```
   q chat "移除 eagleeye-ec-databases 任務狀況確認，最後完成後把 eagleeye-ec-databases & eagleeye-v2-databases 的操作紀錄與關鍵資訊記錄到文件 rds-20251015.md 裡面"
   ```
3. 觀察 CLI 回報 panic 並 crash，而非正常處理並回傳結果。  
4. 若需要更完整的回溯，可設定環境變數並重現：  
   ```
   export RUST_BACKTRACE=1
   q chat "…"
   ```

### Environment

```yaml
<This will be visible to anyone. Do not include personal or sensitive information>

[q-details]
version = "1.18.1"
hash = "5610605baef17b76ef41448f2e943f021c2b2273"
date = "2025-10-13T22:31:07.064465Z (1d ago)"

[system-info]
os = "macOS 26.0.1 (25A362)"
chip = "Apple M1 Pro"
total-cores = 8
memory = "32.00 GB"

[environment]
cwd = "/Users/USER/Library/Mobile Documents/com~apple~CloudDocs/Works/AmazonQ/aws/reddoor/rds"
cli-path = "/Users/USER/Library/Mobile Documents/com~apple~CloudDocs/Works/AmazonQ/aws/reddoor/rds"
install-method = "brew"

[env-vars]
PATH = "/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/Users/USER/.codeium/windsurf/bin:/opt/homebrew/opt/mysql-client/bin:/Users/USER/.local/bin:/Users/USER/.nvm/versions/node/v23.1.0/bin:/Users/USER/.oh-my-zsh/custom/plugins/git-open:/usr/local/opt/coreutils/libexec/gnubin:/Users/USER/bin:/Users/USER/.local/bin:/usr/local/bin:/Applications/Sublime Text.app/Contents/SharedSupport/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Applications/iTerm.app/Contents/Resources/utilities:/Users/USER/.local/bin:/opt/homebrew/opt/fzf/bin:/Users/USER/Library/Python/3.13/bin"
QTERM_SESSION_ID = "8eee7e8886a244ea8bca60eee85faea7"
Q_SET_PARENT_CHECK = "1"
Q_TERM = "1.18.1"
Q_USING_ZSH_AUTOSUGGESTIONS = "1"
SHELL = "/bin/zsh"
TERM = "xterm-256color"
__CFBundleIdentifier = "com.googlecode.iterm2"
```

### PR DESCRIPTION
**Pull request by mzkmnk**
## Summary
Fixes panic when truncating MCP prompt descriptions containing CJK (Chinese, Japanese, Korean) characters.

## Issues
Closes #3117
Closes #3170
Related to #3136, #3086

## Problem
The `truncate_description` function in `prompts.rs` was using unsafe byte-index slicing (`&text[..n]`), which panics when the index falls in the middle of a multibyte UTF-8 character. CJK characters typically use 3 bytes in UTF-8, causing crashes when truncating at byte boundaries.

**Error example:**

byte index 37 is not a char boundary; it is inside '한' (bytes 36..39)

## Solution
- Replaced unsafe byte-index slicing with UTF-8 safe `char_indices()` iteration
- Finds the last valid character boundary before the target length
- Handles edge cases (empty strings, very short max_length, emojis)
- Maintains backward compatibility with ASCII text

## Changes
- Modified `truncate_description()` function in `crates/chat-cli/src/cli/chat/cli/prompts.rs`
- Added comprehensive test cases for CJK characters
- Updated existing tests to reflect UTF-8 safe behavior

## Note

- Cargo.lock was updated to allow local testing and verification of the changes.

## Testing
Verified with test cases from reported issues:
- ✅ Korean text: "사용자가 작성한 글의 어색한 표현이나..." (Issue #3117)
- ✅ Chinese text: "移除 eagleeye-ec-databases 任務狀況確認..." (Issue #3170)
- ✅ Japanese text: "これは日本語のテキストです..."
- ✅ ASCII text (backward compatibility)
- ✅ Edge cases (empty strings, emojis)

All tests pass without panics, respecting character boundaries.

## Impact
- **Scope**: MCP prompt description display (`/prompts list` command)
- **Compatibility**: Fully backward compatible
- **Risk**: Low - only affects truncation logic for long descriptions

スクリーンショット 2025-10-22 午後9 36 39
スクリーンショット 2025-10-22 午後9 37 57

### CODE REVIEW

File: crates/chat-cli/src/cli/chat/cli/prompts.rs [hunk 1]
```diff
@@ -240,12 +240,30 @@ fn format_description(description: Option<&String>) -> String {
...
+            if idx > target_len {
+                break;
+            }
+            truncate_at = idx;
+        }
+        
+        // If we found a valid boundary, use it; otherwise use the last character start
+        if truncate_at == 0 && !text.is_empty() {
+            // Edge case: even the first character is too long
+            truncate_at = text.char_indices().next().map(|(i, _)| i).unwrap_or(0);
```
Comment (evanliu048) [inline crates/chat-cli/src/cli/chat/cli/prompts.rs]: qq: Should we delete this if check? `char_indices().next()` always returns the first character at index 0, so this line doesn't modify `truncate_at`

Comment (mzkmnk) [reply crates/chat-cli/src/cli/chat/cli/prompts.rs]: @evanliu048 

You're absolutely right. Removed in [c029a35](https://github.com/aws/amazon-q-developer-cli/pull/3227/commits/c029a35f7795d3af30584761ecef5280caf22341). All tests still pass. :octocat:

**Comment by mzkmnk** [comment]
@evanliu048 

I've made the corrections. Please review 🙇

File: crates/chat-cli/src/cli/chat/cli/prompts.rs [hunk 2]
```diff
@@ -240,14 +241,14 @@ fn format_description(description: Option<&String>) -> String {
...
 fn truncate_description(text: &str, max_length: usize) -> String {
-    if text.len() <= max_length {
-        text.to_string()
-    } else {
-        let truncated = &text[..max_length.saturating_sub(3)];
-        format!("{}...", truncated.trim_end())
-    }
+    let mut result = text.to_string();
+
+    truncate_safe_in_place(&mut result, max_length, "...");
```
Comment (mzkmnk) [inline crates/chat-cli/src/cli/chat/cli/prompts.rs:254]: I found that `truncate_safe_in_place` already exists in the codebase, so I used that instead :octocat:

**Comment by mzkmnk** [comment]
@evanliu048 

It's difficult to rebase onto main right now, so please review this when you have time.
